### DISCUSSION TITLE
Suggestions for the Search bar

### DISCUSSION THREAD

**Discussion by tmsg-gh** [espanso/espanso#1236] [Feature requests and Ideas]
(Perhaps these things already exist (or are in the pipeline...). If so, ignore :smile: .)

1. Sometimes I want to limit the search to either triggers OR expansions. So ie ":d" might only search within triggers whereas say "!d" might search expansions. Or there could be a new option to show TWO entry fields, one for triggers, the other for expansions. The exact details have to be discussed, of course.
2. The Search Bar could support regex searches or at least some sort of pattern matching (ie anchor search at the beginning or end).
3. Say I have a dozen triggers starting with ":e" and I type ":e" in some app and then open the Search Bar. It would be nice if the Search Bar would only show those triggers/expansions. (I see that this might interfere with how some people use the Search Bar or expect it to work, so perhaps it makes sense to create a second hotkey specifically for that sub-function.)

Thanks for looking into this.

**Comment by federico-terzi** [comment]
Hey @tmsg-gh,

Thank you for the feedback! I'll investigate how we could integrate those features into the search bar without changing the general user experience. I think having a prefix for trigger-only search (as we have `>` for commands) would be reasonable, I'll think about it :)

Cheers :)
